### ISSUE TITLE
istioOperator install always creates istio-operator namespace

### ISSUE THREAD

**Issue by adabuleanu** [istio/istio#24073]
**Bug description**
During IstioOperator install to a custom namespace (--operatorNamespace option), the namespace istio-operator is always created. Instead if the custom namespace does not exist, the operator install gives errors.

**Expected behavior**
During installing the IstioOperator in a custom namespace (--operatorNamespace option), the process must create the custom namespace, not the istio-operator namespace. This leads me to think there is a hard-coded value somewhere.

**Steps to reproduce the bug**
```
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ istioctl version
no running Istio pods in "istio-system"
1.6.0
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get ns
NAME              STATUS   AGE
default           Active   12d
kube-node-lease   Active   12d
kube-public       Active   12d
kube-system       Active   12d
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl create ns istio
namespace/istio created
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get ns
NAME              STATUS   AGE
default           Active   12d
istio             Active   9s
kube-node-lease   Active   12d
kube-public       Active   12d
kube-system       Active   12d
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ istioctl operator init --operatorNamespace istio
Using operator Deployment image: docker.io/istio/operator:1.6.0
- Processing resources for Istio operator.
- Processing resources for Istio operator.
- Processing resources for Istio operator.
  Processing resources for Istio operator.
- Processing resources for Istio operator.
- Processing resources for Istio operator.
- Processing resources for Istio operator.
  Processing resources for Istio operator. Waiting for Deployment/istio/istio-operator
- Processing resources for Istio operator. Waiting for Deployment/istio/istio-operator
✔ Istio operator installed

✔ Installation complete
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get ns
NAME              STATUS   AGE
default           Active   12d
istio             Active   33s
istio-operator    Active   9s
kube-node-lease   Active   12d
kube-public       Active   12d
kube-system       Active   12d
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get all -n istio
NAME                                 READY   STATUS    RESTARTS   AGE
pod/istio-operator-7f9c7ff47-h5qxh   1/1     Running   0          14s

NAME                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/istio-operator   ClusterIP   10.108.188.79   <none>        8383/TCP   14s

NAME                             READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/istio-operator   1/1     1            1           14s

NAME                                       DESIRED   CURRENT   READY   AGE
replicaset.apps/istio-operator-7f9c7ff47   1         1         1       14s
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get all -n istio-operator
No resources found in istio-operator namespace.
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ istioctl operator remove --operatorNamespace istio
Removing Istio operator...
  Pruned object Deployment:istio:istio-operator.
  Pruned object Service:istio:istio-operator.
  Pruned object ServiceAccount:istio:istio-operator.
2020-05-25T09:15:34.332584Z     warn    installer       retrieving resources to prune type networking.istio.io/v1alpha3, Kind=DestinationRule: no matches for kind "DestinationRule" in version "networking.istio.io/v1alpha3" not found
2020-05-25T09:15:34.342916Z     warn    installer       retrieving resources to prune type networking.istio.io/v1alpha3, Kind=EnvoyFilter: no matches for kind "EnvoyFilter" in version "networking.istio.io/v1alpha3" not found
2020-05-25T09:15:34.541405Z     warn    installer       retrieving resources to prune type networking.istio.io/v1alpha3, Kind=Gateway: no matches for kind "Gateway" in version "networking.istio.io/v1alpha3" not found
2020-05-25T09:15:35.281867Z     warn    installer       retrieving resources to prune type networking.istio.io/v1alpha3, Kind=VirtualService: no matches for kind "VirtualService" in version "networking.istio.io/v1alpha3" not found
2020-05-25T09:15:36.021543Z     warn    installer       retrieving resources to prune type security.istio.io/v1beta1, Kind=PeerAuthentication: no matches for kind "PeerAuthentication" in version "security.istio.io/v1beta1" not found
  Pruned object ClusterRole::istio-operator.
  Pruned object ClusterRoleBinding::istio-operator.
Deleted namespace istio
✔ Removal complete
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get ns
NAME              STATUS        AGE
default           Active        12d
istio             Terminating   5m12s
istio-operator    Active        4m48s
kube-node-lease   Active        12d
kube-public       Active        12d
kube-system       Active        12d
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl delete ns istio-operator
namespace "istio-operator" deleted
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl get ns
NAME              STATUS   AGE
default           Active   12d
kube-node-lease   Active   12d
kube-public       Active   12d
kube-system       Active   12d
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ istioctl operator init --operatorNamespace istio
Using operator Deployment image: docker.io/istio/operator:1.6.0
- Processing resources for Istio operator.
- Processing resources for Istio operator.
2020-05-25T09:16:09.815163Z     error   installer       failed to create "ServiceAccount/istio/istio-operator": namespaces "istio" not found
- Processing resources for Istio operator.
  Processing resources for Istio operator.
2020-05-25T09:16:09.911030Z     error   installer       failed to create "Deployment/istio/istio-operator": namespaces "istio" not found
2020-05-25T09:16:09.968678Z     error   installer       failed to create "Service/istio/istio-operator": namespaces "istio" not found
✘ Istio operator encountered an error: failed to create "ServiceAccount/istio/istio-operator": namespaces "istio" not found
failed to create "Deployment/istio/istio-operator": namespaces "istio" not found
failed to create "Service/istio/istio-operator": namespaces "istio" not found

```
**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ istioctl version --remote
no running Istio pods in "istio-system"
1.6.0
[cloud-user@nci-service-mesh-performance-benchmark-1 ~]$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:31:31Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:23:21Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"
```

**How was Istio installed?**
IstioOperator

**Environment where bug was observed (cloud vendor, OS, etc)**
vanilla Kubernetes 1.16.6, Rhel 7.8, Openstack VM

**Comment by kebe7jun** [comment]
@adabuleanu hello, it is a bug, and fixed in #24074

**Comment by adabuleanu** [comment]
with what istio version will the fix be shipped?

**Comment by kebe7jun** [comment]
@adabuleanu This fix has merged into 1.5 and 1.6 release branches.
1.5.5 or 1.6.1 (future) will include this fix.
